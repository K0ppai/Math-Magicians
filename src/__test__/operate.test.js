import operate from '../logic/operate';

describe('test_the_operator_logic', () => {
  // Tests that addition works with positive numbers
  it('test_addition_positive', () => {
    expect(operate(2, 3, '+')).toMatchSnapshot();
  });

  // Tests that subtraction works with positive numbers
  it('test_subtraction_positive', () => {
    expect(operate(5, 3, '-')).toMatchSnapshot();
  });

  // Tests that multiplication works with positive numbers
  it('test_multiplication_positive', () => {
    expect(operate(2, 3, 'x')).toMatchSnapshot();
  });

  // Tests that division by zero returns an error message
  it('test_division_by_zero', () => {
    expect(operate(2, 0, 'รท')).toMatchSnapshot();
  });

  // Tests that addition works with very large numbers
  it('test_addition_very_large_numbers', () => {
    expect(operate('99999999999999999999999999999999999999999999999999', '1', '+')).toMatchSnapshot();
  });

  // Tests that an unknown operation returns an error message
  it('test_unknown_operation', () => {
    expect(() => operate(2, 3, 'z')).toThrowErrorMatchingSnapshot();
  });
});
