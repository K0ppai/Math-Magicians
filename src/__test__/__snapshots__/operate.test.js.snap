// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test_the_operator_logic test_addition_positive 1`] = `"5"`;

exports[`test_the_operator_logic test_addition_very_large_numbers 1`] = `"1e+50"`;

exports[`test_the_operator_logic test_division_by_zero 1`] = `"Can't divide by 0."`;

exports[`test_the_operator_logic test_multiplication_positive 1`] = `"6"`;

exports[`test_the_operator_logic test_subtraction_positive 1`] = `"2"`;

exports[`test_the_operator_logic test_unknown_operation 1`] = `"Unknown operation 'z'"`;
